iOS/Settings/Account and Extensions/Accounts/AccountsManagementView.swift

```diff
@@ -0,0 +1,155 @@
+//
+//  AccountsManagementView.swift
+//  NetNewsWire-iOS
+//
+//  Created by Stuart Breckenridge on 13/11/2022.
+//  Copyright © 2022 Ranchero Software. All rights reserved.
+//
+
+import SwiftUI
+import Account
+import Combine
+
+public final class AccountManagementViewModel: ObservableObject {
+	
+	@Published var sortedActiveAccounts = [Account]()
+	@Published var sortedInactiveAccounts = [Account]()
+	@Published var accountsForDeletion = [Account]()
+	@Published var showAccountDeletionAlert: Bool = false
+	@Published var showAddAccountSheet: Bool = false
+	public var accountToDelete: Account? = nil
+	
+	init() {
+		refreshAccounts()
+		
+		NotificationCenter.default.addObserver(forName: .AccountStateDidChange, object: nil, queue: .main) { [weak self] _ in
+			self?.refreshAccounts()
+		}
+		
+		NotificationCenter.default.addObserver(forName: .UserDidAddAccount, object: nil, queue: .main) { [weak self] _ in
+			self?.refreshAccounts()
+		}
+		
+		NotificationCenter.default.addObserver(forName: .UserDidDeleteAccount, object: nil, queue: .main) { [weak self] _ in
+			self?.refreshAccounts()
+		}
+		
+		NotificationCenter.default.addObserver(forName: .DisplayNameDidChange, object: nil, queue: .main) { [weak self] _ in
+			self?.refreshAccounts()
+		}
```

Comment:
I feel like I'm coming out of nowhere contributing here but every like 6-12  months I find myself coming back to this project and attempting to put some time into it

Anyways, this `NotificationCenter` stuff could use the new [`assign(to:)`](https://developer.apple.com/documentation/combine/fail/assign(to:)) API like so if you're inclined:

```swift
		let changes = Publishers.Merge4(
			NotificationCenter.default.publisher(for: .AccountStateDidChange),
			NotificationCenter.default.publisher(for: .UserDidAddAccount),
			NotificationCenter.default.publisher(for: .UserDidDeleteAccount),
			NotificationCenter.default.publisher(for: .DisplayNameDidChange)
		)
		
		changes
			.map { _ in AccountManager.shared.sortedActiveAccounts }
			.assign(to: &$sortedActiveAccounts)
		
		changes
			.map { _ in AccountManager.shared.sortedAccounts.filter { !$0.isActive} }
			.assign(to: &$sortedInactiveAccounts)
```

Reply:
In fact you could even do away with the initial call to `refreshAccounts()` by doing the following also:
```swift
	init() {
		let changes = Publishers.Merge4(
			NotificationCenter.default.publisher(for: .AccountStateDidChange),
			NotificationCenter.default.publisher(for: .UserDidAddAccount),
			NotificationCenter.default.publisher(for: .UserDidDeleteAccount),
			NotificationCenter.default.publisher(for: .DisplayNameDidChange)
		).share()
		
		changes
			.map { _ in AccountManager.shared.sortedActiveAccounts }
			.prepend(AccountManager.shared.sortedActiveAccounts)
			.assign(to: &$sortedActiveAccounts)
		
		changes
			.map { _ in AccountManager.shared.sortedAccounts.filter { !$0.isActive} }
			.prepend(AccountManager.shared.sortedAccounts.filter { !$0.isActive})
			.assign(to: &$sortedInactiveAccounts)
	}
```

Reply:
> I feel like I'm coming out of nowhere contributing here but every like 6-12 months I find myself coming back to this project and attempting to put some time into it
> 
> Anyways, this `NotificationCenter` stuff could use the new [`assign(to:)`](https://developer.apple.com/documentation/combine/fail/assign(to:)) API like so if you're inclined:
> 
> ```swift
> 		let changes = Publishers.Merge4(
> 			NotificationCenter.default.publisher(for: .AccountStateDidChange),
> 			NotificationCenter.default.publisher(for: .UserDidAddAccount),
> 			NotificationCenter.default.publisher(for: .UserDidDeleteAccount),
> 			NotificationCenter.default.publisher(for: .DisplayNameDidChange)
> 		).share()
> 		
> 		changes
> 			.map { _ in AccountManager.shared.sortedActiveAccounts }
> 			.assign(to: &$sortedActiveAccounts)
> 		
> 		changes
> 			.map { _ in AccountManager.shared.sortedAccounts.filter { !$0.isActive} }
> 			.assign(to: &$sortedInactiveAccounts)
> ```

We don't use the Combine framework in that way anywhere else in the app. It would be inconsistent to use it here.

Reply:
No worries, it may be inconsistent but so is SwiftUI, new shift could be good for new paradigm. Sometimes it makes sense to use the new APIs. 

In this specific instance you actually don't need to concern about reference cycles, reducing room for error.

Reply:
I think the closest _new_ thing to use would `AsyncSequence`. For example, in `MasterFeedController`, there are eight `NotificationCenter.default.addObserver`s in `viewDidLoad`. In `UIKit`, they'd be replaced:

```swift

private var notificationsTask: Task<(), Never>?

override func viewDidLoad() {

    // NotificationCenter.default.addObserver x8 removed
    
    notificationsTask = Task {
        await notificationHandler()
    }

}


func notificationHandler() async {
    for await notification in NotificationCenter.default.notifications(named: .UnreadCountDidChange) {
        unreadCountDidChange(notification)
    }
    
    // ... and the rest
}
```

You could then get rid of `@objc` from the method signatures. 

In `SwiftUI`, the above is simpler, as you don't need to worry about retain cycles when views are removed. The `notificationHander` function can be added to a `.task` modifier.

Interesting architectural discussion for @brentsimmons  / @vincode-io.